src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -485,45 +458,48 @@ def _linear_acceleration_of_collidable_points(
         CW_J_WC_BW: jtp.MatrixLike,
         CW_J_dot_WC_BW: jtp.MatrixLike,
     ) -> jtp.Matrix:
-        CW_J̇_WC_BW = CW_J_dot_WC_BW
+
         BW_ν = BW_nu
         BW_ν̇ = BW_nu_dot
+        CW_J̇_WC_BW = CW_J_dot_WC_BW
 
+        # Compute the linear acceleration of the collidable points.
+        # Since we use doubly-mixed jacobians, this corresponds to W_p̈_C.
         CW_a_WC = jnp.vstack(CW_J̇_WC_BW) @ BW_ν + jnp.vstack(CW_J_WC_BW) @ BW_ν̇
-        CW_a_WC = CW_a_WC.reshape(-1, 6)
 
+        CW_a_WC = CW_a_WC.reshape(-1, 6)
         return CW_a_WC[:, 0:3].squeeze()
 
     @staticmethod
     def _compute_baumgarte_stabilization_term(
         inactive_collidable_points: jtp.ArrayLike,
-        h: jtp.ArrayLike,
-        ḣ: jtp.ArrayLike,
+        δ: jtp.ArrayLike,
+        δ̇: jtp.ArrayLike,
         K: jtp.FloatLike,
         D: jtp.FloatLike,
     ) -> jtp.Array:
-        def baumgarte_stabilization(
+
+        def baumgarte_stabilization_of_single_point(
             inactive: jtp.BoolLike,
-            h: jtp.FloatLike,
-            ḣ: jtp.FloatLike,
+            δ: jtp.FloatLike,
+            δ̇: jtp.FloatLike,
             k_baumgarte: jtp.FloatLike,
             d_baumgarte: jtp.FloatLike,
         ) -> jtp.Array:
+
             baumgarte_term = jax.lax.cond(
                 inactive,
-                lambda h, ḣ, K, D: jnp.zeros(shape=(3,)),
-                lambda h, ḣ, K, D: jnp.zeros(shape=(3,)).at[2].set(K * h + D * ḣ),
-                *(
-                    h,
-                    ḣ,
-                    k_baumgarte,
-                    d_baumgarte,
-                ),
+                lambda δ, δ̇̇, K, D: jnp.zeros(3),
+                # This is equivalent to: K*(pT - p)⋅n̂ + D*(0 - v)⋅n̂,
+                # where pT is the point on the terrain surface vertical to p.
+                lambda δ, δ̇, K, D: jnp.zeros(3).at[2].set(K * δ + D * δ̇),
+                *(δ, δ̇, k_baumgarte, d_baumgarte),
```

Comment:
As discussed f2f, I have the impression that here there was an error in the sign of the $D \dot{\mathbf{h}}$ term.

I refactored the code to use the normal material deformation $\delta$ instead of the penetration depth $\mathbf{h}$, that are related as follows:

```math
\begin{cases}
\delta &= \mathbf{h} \cdot \hat{\boldsymbol{n}} = -(\mathcal{H}(\mathbf{p}) - \mathbf{p}) \cdot \hat{\boldsymbol{n}} \\
\dot{\delta} &= (\dot{\mathbf{p}} \cdot \hat{\boldsymbol{n}}) \: \hat{\boldsymbol{n}}
\end{cases}
```

Reply:
Nice catch thanks! It seems sounds to me